lib/libc/arcmwdt/threading.c

```diff
@@ -6,22 +6,24 @@
 
 #ifdef CONFIG_MULTITHREADING
 
+#include <stdio.h>
 #include <zephyr/init.h>
 #include <zephyr/kernel.h>
 #include <zephyr/sys/__assert.h>
 #include <zephyr/sys/mutex.h>
 #include <zephyr/logging/log.h>
-#include <../lib/src/c/inc/internal/thread.h>
 
 #ifndef CONFIG_USERSPACE
 #define ARCMWDT_DYN_LOCK_SZ	(sizeof(struct k_mutex))
-#define ARCMWDT_MAX_DYN_LOCKS	10
+#define ARCMWDT_MAX_DYN_LOCKS	(FOPEN_MAX * 2 + 10)
```

Comment:
Probably we can lower down this '10' as file locks were previously used from this pool.

And it is worth adding a comment here about 
> The library wants to have 2 locks per each available FILE entry

Reply:
Do you know the exact number on top of the file locks? I've seen 3, but there may be more. Will 5 be enough?